katdal/sensordata.py

```diff
@@ -126,34 +186,112 @@ class TelstateSensorData(SensorData):
     Parameters
     ----------
     data : recarray-like, with fields ('timestamp', 'value')
-        Uncached sensor data as record array or equivalent (such as a
-        :class:`h5py.Dataset`)
+        Uninterpolated sensor data as structured array or equivalent (such as
+        an :class:`h5py.Dataset`)
```

Comment:
Extraction is the combination of

- *getting* the raw sensor data from the store (telstate / HDF5),
- *interpolating* it onto the correlator data timestamps, and
- optionally *selecting* bits of it based on the currently active selection in the dataset.

The interpolated output (categorical or numerical) replaces the raw SensorData object in the SensorCache, meaning extraction typically only happens once per sensor.